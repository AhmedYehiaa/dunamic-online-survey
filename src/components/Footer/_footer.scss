.footer {
  display: flex;
  justify-content: space-between;
  margin: 30px auto 0;
  max-width: 75%; 
}