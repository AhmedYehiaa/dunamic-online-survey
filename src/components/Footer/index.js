import React from 'react';
import Button, { TYPES, SIZES } from '../Button';

const Footer = () => (
  <footer className="footer">
    <div className="button-back">
      <Button
        type="button"
        onClick={() => { console.log('BACK'); }}
        text="Back"
        disabled={false}
        buttonType={TYPES.BACK}
        buttonSize={SIZES.MEDIUM}
      />
    </div>
    <div className="button-next">
      <Button
        type="button"
        onClick={() => { console.log('NEXT'); }}
        text="Next"
        disabled={false}
        buttonType={TYPES.PRIMARY}
        buttonSize={SIZES.MEDIUM}
      />
    </div>
  </footer>
);
export default Footer;
