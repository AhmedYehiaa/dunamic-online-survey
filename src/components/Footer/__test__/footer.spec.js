import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../index';

it('it should renders the footer component', () => {
  const handleBackNavigation = jest.fn();
  const submit = jest.fn();

  const wrapper = shallow(
    <Footer
      disableBackButton={false}
      disableNextButton={false}
      onClickBack={handleBackNavigation}
      onClickNext={submit}
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
