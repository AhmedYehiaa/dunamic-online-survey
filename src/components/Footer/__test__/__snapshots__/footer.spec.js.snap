// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the footer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    disableBackButton={false}
    disableNextButton={false}
    onClickBack={[MockFunction]}
    onClickNext={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button-back"
        >
          <Button
            buttonSize="btn-medium"
            buttonType="btn-back"
            disabled={false}
            onClick={[MockFunction]}
            text="Back"
            type="button"
          />
        </div>,
        <div
          className="button-next"
        >
          <Button
            buttonSize="btn-medium"
            buttonType="btn-primary"
            disabled={false}
            onClick={[MockFunction]}
            text="Next"
            type="button"
          />
        </div>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            buttonSize="btn-medium"
            buttonType="btn-back"
            disabled={false}
            onClick={[MockFunction]}
            text="Back"
            type="button"
          />,
          "className": "button-back",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonSize": "btn-medium",
            "buttonType": "btn-back",
            "disabled": false,
            "onClick": [MockFunction],
            "text": "Back",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            buttonSize="btn-medium"
            buttonType="btn-primary"
            disabled={false}
            onClick={[MockFunction]}
            text="Next"
            type="button"
          />,
          "className": "button-next",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonSize": "btn-medium",
            "buttonType": "btn-primary",
            "disabled": false,
            "onClick": [MockFunction],
            "text": "Next",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button-back"
          >
            <Button
              buttonSize="btn-medium"
              buttonType="btn-back"
              disabled={false}
              onClick={[MockFunction]}
              text="Back"
              type="button"
            />
          </div>,
          <div
            className="button-next"
          >
            <Button
              buttonSize="btn-medium"
              buttonType="btn-primary"
              disabled={false}
              onClick={[MockFunction]}
              text="Next"
              type="button"
            />
          </div>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              buttonSize="btn-medium"
              buttonType="btn-back"
              disabled={false}
              onClick={[MockFunction]}
              text="Back"
              type="button"
            />,
            "className": "button-back",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonSize": "btn-medium",
              "buttonType": "btn-back",
              "disabled": false,
              "onClick": [MockFunction],
              "text": "Back",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              buttonSize="btn-medium"
              buttonType="btn-primary"
              disabled={false}
              onClick={[MockFunction]}
              text="Next"
              type="button"
            />,
            "className": "button-next",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonSize": "btn-medium",
              "buttonType": "btn-primary",
              "disabled": false,
              "onClick": [MockFunction],
              "text": "Next",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
